### PR TITLE
docs: adding markup reference docs

### PR DESCRIPTION
**Pull request by dansiegel**
## PR Type

What kind of change does this PR introduce?

- Documentation content changes

Docs for C# Markup

### CODE REVIEW

**Review by nickrandolph — CHANGES_REQUESTED** [review]
Add Uid to each page (eg https://github.com/unoplatform/uno.extensions/blob/2b1d284d7daae3dddf9495b583ad92f31f922d1b/doc/Reference/Navigation/Navigation.md?plain=1#L2)
For references use Uid instead of filename (eg https://github.com/unoplatform/uno.extensions/blob/2b1d284d7daae3dddf9495b583ad92f31f922d1b/doc/Learn/Tutorials/Navigation/HowTo-DisplayItem.md?plain=1#L9)

File: doc/Reference/Markup/Templates.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+		.Child(new ContentPresenter()
+			.Content(x => x.TemplateBind(() => button.Content))
+```
+
+---
+uid: Reference.Markup.Templates
+---
+## All other Framework Templates
+
+In all other cases Template properties are treated the same with a simple delegate expected to return the desired UI. In these cases it is assumed that you have the same context as the parent and have no need for additional typing on the delegate.
```
Comment (jeromelaban) [inline doc/Reference/Markup/Templates.md:51]:
```suggestion
In all other cases, `Template` properties are treated the same with a simple delegate expected to return the desired UI. In these cases, it is assumed that the same context as the parent is available and there is no need for additional typing on the delegate.
```

File: doc/Reference/Markup/VisualStateManager.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
+---
+uid: Reference.Markup.VisualStateManager
+---
+# Visual State Manager
+
+Similar to other API's the Visual State Manager provides a fluent Builder API to help you add one or more groups each with one or more states. In it's simplest form you can add a group with a single state like:
```
Comment (jeromelaban) [inline doc/Reference/Markup/VisualStateManager.md:6]:
```suggestion
Similar to other API's the `VisualStateManager` provides a fluent Builder API to help adding one or more groups, each with one or more states. In its simplest form, adding a group with a single state looks like this:
```

File: doc/Reference/Markup/VisualStateManager.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+new Grid()
+	.VisualStateManager(vsm => vsm
+		.Group(group => group
+			.State("SomeState", state => state
+				.Setters(grid => grid.Background(StaticResource.Get<Brush>("SomeResource"))))))
+```
+
+## Updating Children
+
+When updating the children of a given control it is important to ensure that these children exist in scope. You can provide a reference to a given control instance for the Setters to update setters specific to that control.
```
Comment (jeromelaban) [inline doc/Reference/Markup/VisualStateManager.md:18]:
```suggestion
When updating the children of a given control, it is important to ensure that these children exist in scope. You can provide a reference to a given control instance for the `Setters` to update those specific to the current instant.
```

File: doc/Reference/Markup/VisualStateManager.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+	.VisualStateManager(vsm => vsm
+		.Group(group => group
+			.State("SomeState", state => state
+				.Setters(grid => grid.Background(StaticResource.Get<Brush>("SomeResource")))
+				.Setters(textBlock, tb => tb.Text("You are in Some State")))));
+```
+
+## Using Triggers
+
+In addition to adding various style setters to a given state you may also add one or more Triggers to your Visual State.
```
Comment (jeromelaban) [inline doc/Reference/Markup/VisualStateManager.md:34]:
```suggestion
In addition to adding various style setters to a given state, it is also possible to add one or more `Triggers` to a Visual State.
```

File: doc/Reference/Markup/DependencyPropertyBuilder.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+new TextBlock()
+	.Text(x => x.Bind(() => vm.Message))
+```
+
+When creating a simplified binding such as the one above you may alternatively simply provide the binding expression like:
+
+```cs
+new TextBlock()
+	.Text(() => vm.Message);
+```
```
Comment (jeromelaban) [inline doc/Reference/Markup/DependencyPropertyBuilder.md:22]: We should give an example on using complex path expressions

File: doc/Reference/Markup/GettingStarted.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
+---
+uid: Reference.Markup.GettingStarted
+---
+# Getting Started
+
+Uno.Extensions.Markup is a collection of packages to make it easier for developers to build UI for their Uno and WinUI applications using a fluent and entirely C# based approach. You may want to add one or more packages based on your needs for your application.
+
+- **Uno.Extensions.Markup** - This is the core package that contains the base helpers that make C# Markup possible.
```
[the remark this answers is not in the document]
Comment (dansiegel) [reply doc/Reference/Markup/GettingStarted.md:8]: true probably should reference this more in the Generating docs

File: doc/Reference/Markup/Binding101.md [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+
+```cs
+public partial class MainPage : Page
+{
+    public MainPage()
+    {
+        this.DataContext<MyViewModel>((page, vm) => page
+            .Content(new TextBlock()
+                .Text(() => vm.Query)
+                .Convert(query => $"Search: {query}")
```
Comment (agneszitte-nventive) [inline doc/Reference/Markup/Binding101.md:78]: Should be within the Text(....) extension here ;)

File: doc/Reference/Markup/Binding101.md [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+```cs
+public partial class MainPage : Page
+{
+    public MainPage()
+    {
+        this.DataContext<MyViewModel>((page, vm) => page
+            .Content(new TextBox()
+                .Text(() => vm.Enabled)
+                .Convert(enabled => $"Enabled: {enabled}")
+                .ConvertBack(text => bool.TryParse(text.Replace("Search: ", ""), out var enabled) ? enabled : false)
```
Comment (agneszitte-nventive) [inline doc/Reference/Markup/Binding101.md:95]: Should be within the Text(....) extension here ;)

File: doc/Reference/Markup/Binding101.md [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+public partial class MainPage : Page
+{
+    public MainPage()
+    {
+        this.DataContext<MyViewModel>((page, vm) => page
+            .Content(
+                new MyCustomControl()
+                    .DataContext(x => x
+                        .Bind(() => this.DataContext)
+                        .Convert(dataContext => {
```
Comment (agneszitte-nventive) [inline doc/Reference/Markup/Binding101.md:137]: Should be within the Bind(....) extension here ;)

File: doc/Reference/Markup/DependencyPropertyBuilder.md [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+
+### Converters
+
+The `DependencyPropertyBuilder`'s Bind method will return a strongly typed `BindingFactory`. This can help you as you tackle more advanced scenarios for Bindings such as by providing strongly typed converters that are picked up by the combination of the expected property type and the property type of the bound property. For instance you might have something like:
+
+```cs
+new TextBox()
+	.Text(() => vm.Query)
+	.Foreground(x => x.Bind(() => vm.Query)
+		.Convert(query => new SolidColorBrush(!string.IsNullOrEmpty(query) && query.Length > 5 ? Colors.Green : Colors.Red)));
```
Comment (agneszitte-nventive) [inline doc/Reference/Markup/DependencyPropertyBuilder.md:52]: Should be within the Foreground(....) here ;)

Comment (dansiegel) [reply doc/Reference/Markup/DependencyPropertyBuilder.md:52]: it is within the Foreground

File: doc/Reference/Markup/Styles.md [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+```cs
+new Style<Button>()
+	.Setters(s => s.Background(StaticResource.Get<Brush>("MyBrush")))
+```
+
+Alternatively if you are using a Theme Resource from Uno.Material you can use Uno.Themes.WinUI.Markup to use a Theme Resource like:
+
+```cs
+new Style<Button>()
+	.Setters(Theme.Brushes.Primary.Default)
```
Comment (agneszitte-nventive) [inline doc/Reference/Markup/Styles.md:45]:
```suggestion
	.Setters(s => s.Background(Theme.Brushes.Primary.Default))
```
Missing a specific property in the snippet for this setter

**Pull request closed without merging**